### ISSUE TITLE
[REGRESSION]: PublishCodeCoverageResults@2 no longer uploads the artifacts to the pipeline

### ISSUE THREAD

**Issue by carboneater** [microsoft/azure-pipelines-tasks#19547]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

PublishCodeCoverageResults@2

### Breaking task version

2.235.1

### Last working task version

2.234.0

### Regression Description

PublishCodeCoverageResults@2 <= 2.234.0 would upload the coverage reports as pipeline artifacts.

With 2.235.1, although the `Code Coverage` tab is populated in devops, there are no coverage artifacts in the pipeline run.

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Ubuntu 20.04

### Relevant log output

```shell
There are no errors reported by the Task.
Although later jobs trying to access the coverage artifacts DO fail as they're missing the report.
```

### Full task logs with system.debug enabled

<details>
  <summary>UNSUCCESSFUL RUN</summary>
  <pre>2024-02-14T20:40:48.3634808Z ##[debug]Evaluating condition for step: 'PublishCodeCoverageResults'
2024-02-14T20:40:48.3635934Z ##[debug]Evaluating: and(succeeded(), eq('True', 'true'))
2024-02-14T20:40:48.3636299Z ##[debug]Evaluating and:
2024-02-14T20:40:48.3636505Z ##[debug]..Evaluating succeeded:
2024-02-14T20:40:48.3636917Z ##[debug]..=> True
2024-02-14T20:40:48.3637106Z ##[debug]..Evaluating eq:
2024-02-14T20:40:48.3637278Z ##[debug]....Evaluating String:
2024-02-14T20:40:48.3637483Z ##[debug]....=> 'True'
2024-02-14T20:40:48.3637645Z ##[debug]....Evaluating String:
2024-02-14T20:40:48.3637819Z ##[debug]....=> 'true'
2024-02-14T20:40:48.3637953Z ##[debug]..=> True
2024-02-14T20:40:48.3638090Z ##[debug]=> True
2024-02-14T20:40:48.3638437Z ##[debug]Expanded: and(True, eq('True', 'true'))
2024-02-14T20:40:48.3638657Z ##[debug]Result: True
2024-02-14T20:40:48.3638924Z ##[section]Starting: PublishCodeCoverageResults
2024-02-14T20:40:48.3642880Z ==============================================================================
2024-02-14T20:40:48.3643051Z Task         : Publish code coverage results v2
2024-02-14T20:40:48.3643133Z Description  : Publish any of the code coverage results from a build
2024-02-14T20:40:48.3643245Z Version      : 2.235.1
2024-02-14T20:40:48.3643323Z Author       : Microsoft Corporation
2024-02-14T20:40:48.3643399Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-code-coverage-results
2024-02-14T20:40:48.3643531Z ==============================================================================
2024-02-14T20:40:48.4473500Z ##[debug]Using node path: /home/vsts/agents/3.234.0/externals/node20_1/bin/node
2024-02-14T20:40:48.5386079Z ##[debug]agent.TempDirectory=/home/vsts/work/_temp
2024-02-14T20:40:48.5399083Z ##[debug]loading inputs and endpoints
2024-02-14T20:40:48.5403289Z ##[debug]loading INPUT_SUMMARYFILELOCATION
2024-02-14T20:40:48.5420940Z ##[debug]loading INPUT_FAILIFCOVERAGEEMPTY
2024-02-14T20:40:48.5421790Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2024-02-14T20:40:48.5423646Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2024-02-14T20:40:48.5425313Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2024-02-14T20:40:48.5434256Z ##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
2024-02-14T20:40:48.5436520Z ##[debug]loaded 6
2024-02-14T20:40:48.5441745Z ##[debug]Agent.ProxyUrl=undefined
2024-02-14T20:40:48.5442906Z ##[debug]Agent.CAInfo=undefined
2024-02-14T20:40:48.5443265Z ##[debug]Agent.ClientCert=undefined
2024-02-14T20:40:48.5444432Z ##[debug]Agent.SkipCertValidation=undefined
2024-02-14T20:40:48.5455997Z ##[debug]check path : /home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/task.json
2024-02-14T20:40:48.5459853Z ##[debug]adding resource file: /home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/task.json
2024-02-14T20:40:48.5460302Z ##[debug]system.culture=en-US
2024-02-14T20:40:48.5472279Z ##[debug]summaryFileLocation=coverage.xml
2024-02-14T20:40:48.5475581Z ##[debug]failIfCoverageEmpty=false
2024-02-14T20:40:48.5476080Z ##[debug]System.DefaultWorkingDirectory=/home/vsts/work/1/s
2024-02-14T20:40:48.5477049Z ##[debug]pathToSources=undefined
2024-02-14T20:40:48.5482362Z ##[debug]defaultRoot: '/home/vsts/work/1/s'
2024-02-14T20:40:48.5482990Z ##[debug]findOptions.allowBrokenSymbolicLinks: 'false'
2024-02-14T20:40:48.5483386Z ##[debug]findOptions.followSpecifiedSymbolicLink: 'false'
2024-02-14T20:40:48.5483909Z ##[debug]findOptions.followSymbolicLinks: 'false'
2024-02-14T20:40:48.5484354Z ##[debug]findOptions.skipMissingFiles: 'undefined'
2024-02-14T20:40:48.5485758Z ##[debug]matchOptions.debug: 'false'
2024-02-14T20:40:48.5486191Z ##[debug]matchOptions.nobrace: 'true'
2024-02-14T20:40:48.5486800Z ##[debug]matchOptions.noglobstar: 'false'
2024-02-14T20:40:48.5487777Z ##[debug]matchOptions.dot: 'true'
2024-02-14T20:40:48.5488154Z ##[debug]matchOptions.noext: 'false'
2024-02-14T20:40:48.5489860Z ##[debug]matchOptions.nocase: 'false'
2024-02-14T20:40:48.5490226Z ##[debug]matchOptions.nonull: 'false'
2024-02-14T20:40:48.5491003Z ##[debug]matchOptions.matchBase: 'false'
2024-02-14T20:40:48.5491580Z ##[debug]matchOptions.nocomment: 'false'
2024-02-14T20:40:48.5492006Z ##[debug]matchOptions.nonegate: 'false'
2024-02-14T20:40:48.5492441Z ##[debug]matchOptions.flipNegate: 'false'
2024-02-14T20:40:48.5493333Z ##[debug]pattern: 'coverage.xml'
2024-02-14T20:40:48.5506742Z ##[debug]findPath: '/home/vsts/work/1/s/coverage.xml'
2024-02-14T20:40:48.5507127Z ##[debug]statOnly: 'true'
2024-02-14T20:40:48.5507741Z ##[debug]found 1 paths
2024-02-14T20:40:48.5508137Z ##[debug]applying include pattern
2024-02-14T20:40:48.5509215Z ##[debug]adjustedPattern: '/home/vsts/work/1/s/coverage.xml'
2024-02-14T20:40:48.5514160Z ##[debug]1 matches
2024-02-14T20:40:48.5514880Z ##[debug]1 final results
2024-02-14T20:40:48.5530944Z Found 1 result(s) matching pattern: coverage.xml
2024-02-14T20:40:48.5533062Z /home/vsts/work/1/s/coverage.xml
2024-02-14T20:40:48.5535702Z ##[debug]Agent.Version=3.234.0
2024-02-14T20:40:48.5536214Z ##[debug]Agent.TempDirectory=/home/vsts/work/_temp
2024-02-14T20:40:48.5542055Z ##[debug]which 'dotnet'
2024-02-14T20:40:48.5547813Z ##[debug]found: '/usr/bin/dotnet'
2024-02-14T20:40:48.5627538Z ##[debug]which '/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console'
2024-02-14T20:40:48.5628620Z ##[debug]found: '/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console'
2024-02-14T20:40:48.5629543Z ##[debug]/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console arg: /home/vsts/work/1/s/coverage.xml
2024-02-14T20:40:48.5630494Z ##[debug]/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console arg: --reportDirectory
2024-02-14T20:40:48.5631653Z ##[debug]/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console arg: /home/vsts/work/_temp/3d03ace6-7b7c-4733-a6c5-33934a319ab6
2024-02-14T20:40:48.5633762Z ##[debug]SystemVssConnection auth param AccessToken = ***
2024-02-14T20:40:48.5634529Z ##[debug]System.TeamFoundationCollectionUri=https://dev.azure.com/[REDACTED]/
2024-02-14T20:40:48.5635404Z ##[debug]Build.BuildId=[REDACTED]
2024-02-14T20:40:48.5635776Z ##[debug]Build.ContainerId=26560550
2024-02-14T20:40:48.5636118Z ##[debug]Agent.TempPath=undefined
2024-02-14T20:40:48.5636481Z ##[debug]System.TeamProjectId=[REDACTED]
2024-02-14T20:40:48.5636847Z ##[debug]PIPELINES_COVERAGEPUBLISHER_DEBUG=undefined
2024-02-14T20:40:48.5637345Z ##[debug]exec tool: /home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console
2024-02-14T20:40:48.5637825Z ##[debug]arguments:
2024-02-14T20:40:48.5638149Z ##[debug]   /home/vsts/work/1/s/coverage.xml
2024-02-14T20:40:48.5638478Z ##[debug]   --reportDirectory
2024-02-14T20:40:48.5638830Z ##[debug]   /home/vsts/work/_temp/3d03ace6-7b7c-4733-a6c5-33934a319ab6
2024-02-14T20:40:48.5639895Z [command]/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console /home/vsts/work/1/s/coverage.xml --reportDirectory /home/vsts/work/_temp/3d03ace6-7b7c-4733-a6c5-33934a319ab6
2024-02-14T20:40:49.9912323Z ##[debug]Agent environment resources - Disk: / Available 18750.00 MB out of 74244.00 MB, Memory: Used 656.00 MB out of 6921.00 MB, CPU: Usage 7.27%
2024-02-14T20:40:49.9962542Z ##[debug]PERF : Parser.Parsing : took 0.364 ms.
2024-02-14T20:40:49.9992191Z ##[debug]Publishing file json coverage supported.
2024-02-14T20:40:50.0255293Z ##[debug]Publishing native coverage files is supported.
2024-02-14T20:40:50.0276242Z Publishing file coverage data.
2024-02-14T20:40:50.0277037Z ##[debug]PERF : AzurePipelinesPublisher.UploadNativeCoverageFiles : took 0.1048 ms.
2024-02-14T20:40:50.0289301Z ##[debug]ReportGeneratorTool.ParseCoverageFiles: Parsing coverage files.
2024-02-14T20:40:50.0488477Z 2024-02-14T20:40:50: Loading report '/home/vsts/work/1/s/coverage.xml' 1/1 in memory
2024-02-14T20:40:50.0734260Z 2024-02-14T20:40:50: Preprocessing report
2024-02-14T20:40:50.0758551Z 2024-02-14T20:40:50: Initiating parser for Cobertura
2024-02-14T20:40:50.0863636Z 2024-02-14T20:40:50: Current Assembly: fn_jira
2024-02-14T20:40:50.1020566Z 2024-02-14T20:40:50: Current Assembly: fn_jira.components
2024-02-14T20:40:50.1065382Z 2024-02-14T20:40:50: Current Assembly: fn_jira.poller
2024-02-14T20:40:50.1071122Z 2024-02-14T20:40:50: Current Assembly: fn_jira.util
2024-02-14T20:40:50.1098410Z 2024-02-14T20:40:50: Finished parsing '/home/vsts/work/1/s/coverage.xml' 1/1
2024-02-14T20:40:50.1099123Z 2024-02-14T20:40:50: Parsing of 1 files completed
2024-02-14T20:40:50.1099559Z 2024-02-14T20:40:50: Starting merging result 1
2024-02-14T20:40:50.1114873Z 2024-02-14T20:40:50: Finished merging result 1
2024-02-14T20:40:50.1131018Z ##[debug]ReportGeneratorTool.CreateHTMLReportFromParserResult: Creating HTML report.
2024-02-14T20:40:50.1417770Z 2024-02-14T20:40:50: Initializing report builders for report types: HtmlInline_AzurePipelines
2024-02-14T20:40:50.1492851Z 2024-02-14T20:40:50: Analyzing 15 classes
2024-02-14T20:40:50.1497296Z 2024-02-14T20:40:50: Creating report 1/15 (Assembly: fn_jira, Class: __init__.py)
2024-02-14T20:40:50.1665883Z 2024-02-14T20:40:50: Writing report file 'fn_jira_py.html'
2024-02-14T20:40:50.1891634Z 2024-02-14T20:40:50: Creating report 2/15 (Assembly: fn_jira.components, Class: __init__.py)
2024-02-14T20:40:50.1897964Z 2024-02-14T20:40:50: Writing report file 'fn_jira.components_py.html'
2024-02-14T20:40:50.1981335Z 2024-02-14T20:40:50: Creating report 3/15 (Assembly: fn_jira.components, Class: jira_create_comment.py)
2024-02-14T20:40:50.1990272Z 2024-02-14T20:40:50: Writing report file 'fn_jira.components_py.2.html'
2024-02-14T20:40:50.2016331Z 2024-02-14T20:40:50: Creating report 4/15 (Assembly: fn_jira.components, Class: jira_transition_issue.py)
2024-02-14T20:40:50.2026136Z 2024-02-14T20:40:50: Writing report file 'fn_jira.components_py.3.html'
2024-02-14T20:40:50.2037537Z 2024-02-14T20:40:50: Creating report 5/15 (Assembly: fn_jira.poller, Class: __init__.py)
2024-02-14T20:40:50.2046172Z 2024-02-14T20:40:50: Writing report file 'fn_jira.poller_py.html'
2024-02-14T20:40:50.2053918Z 2024-02-14T20:40:50: Creating report 6/15 (Assembly: fn_jira.poller, Class: configure_tab.py)
2024-02-14T20:40:50.2062255Z 2024-02-14T20:40:50: Writing report file 'fn_jira.poller_py.2.html'
2024-02-14T20:40:50.2074546Z 2024-02-14T20:40:50: Creating report 7/15 (Assembly: fn_jira.poller, Class: poller.py)
2024-02-14T20:40:50.2084970Z 2024-02-14T20:40:50: Writing report file 'fn_jira.poller_py.3.html'
2024-02-14T20:40:50.2142916Z 2024-02-14T20:40:50: Creating report 8/15 (Assembly: fn_jira.util, Class: __init__.py)
2024-02-14T20:40:50.2163720Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.html'
2024-02-14T20:40:50.2177801Z 2024-02-14T20:40:50: Creating report 9/15 (Assembly: fn_jira.util, Class: config.py)
2024-02-14T20:40:50.2183439Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.2.html'
2024-02-14T20:40:50.2214656Z 2024-02-14T20:40:50: Creating report 10/15 (Assembly: fn_jira.util, Class: customize.py)
2024-02-14T20:40:50.2219525Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.3.html'
2024-02-14T20:40:50.2238664Z 2024-02-14T20:40:50: Creating report 11/15 (Assembly: fn_jira.util, Class: helper.py)
2024-02-14T20:40:50.2244760Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.4.html'
2024-02-14T20:40:50.2264280Z 2024-02-14T20:40:50: Creating report 12/15 (Assembly: fn_jira.util, Class: jira.py)
2024-02-14T20:40:50.2272090Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.5.html'
2024-02-14T20:40:50.2293430Z 2024-02-14T20:40:50: Creating report 13/15 (Assembly: fn_jira.util, Class: regex.py)
2024-02-14T20:40:50.2296775Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.6.html'
2024-02-14T20:40:50.2313737Z 2024-02-14T20:40:50: Creating report 14/15 (Assembly: fn_jira.util, Class: selftest.py)
2024-02-14T20:40:50.2319233Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.7.html'
2024-02-14T20:40:50.2337669Z 2024-02-14T20:40:50: Creating report 15/15 (Assembly: fn_jira.util, Class: typings.py)
2024-02-14T20:40:50.2342515Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.8.html'
2024-02-14T20:40:50.2368156Z 2024-02-14T20:40:50: Creating summary
2024-02-14T20:40:50.2422257Z 2024-02-14T20:40:50: Writing report file '/home/vsts/work/_temp/3d03ace6-7b7c-4733-a6c5-33934a319ab6/index.html'
2024-02-14T20:40:50.2753955Z ##[debug]Parser.GenerateHTMLReport: Creating summary file directory: /home/vsts/work/_temp/3d03ace6-7b7c-4733-a6c5-33934a319ab6/Summary_5da2e1d2
2024-02-14T20:40:50.2755231Z ##[debug]Parser.GenerateHTMLReport: Copying summary file /home/vsts/work/1/s/coverage.xml
2024-02-14T20:40:50.2757632Z ##[debug]PERF : Parser.ReportGeneration : took 163.2996 ms.
2024-02-14T20:40:50.2769442Z ##[debug]ReportGeneratorTool.GetFileCoverageInfos: Generating file coverage info from coverage files.
2024-02-14T20:40:50.2797099Z ##[debug]ReportGeneratorTool.CreateHTMLReportFromParserResult: Creating HTML report.
2024-02-14T20:40:50.2805433Z 2024-02-14T20:40:50: Initializing report builders for report types: HtmlInline_AzurePipelines
2024-02-14T20:40:50.2811279Z 2024-02-14T20:40:50: Analyzing 15 classes
2024-02-14T20:40:50.2813525Z 2024-02-14T20:40:50: Creating report 1/15 (Assembly: fn_jira, Class: __init__.py)
2024-02-14T20:40:50.2820247Z 2024-02-14T20:40:50: Writing report file 'fn_jira_py.html'
2024-02-14T20:40:50.2842300Z 2024-02-14T20:40:50: Creating report 2/15 (Assembly: fn_jira.components, Class: __init__.py)
2024-02-14T20:40:50.2848247Z 2024-02-14T20:40:50: Writing report file 'fn_jira.components_py.html'
2024-02-14T20:40:50.2873329Z 2024-02-14T20:40:50: Creating report 3/15 (Assembly: fn_jira.components, Class: jira_create_comment.py)
2024-02-14T20:40:50.2881146Z 2024-02-14T20:40:50: Writing report file 'fn_jira.components_py.2.html'
2024-02-14T20:40:50.2902953Z 2024-02-14T20:40:50: Creating report 4/15 (Assembly: fn_jira.components, Class: jira_transition_issue.py)
2024-02-14T20:40:50.2907177Z 2024-02-14T20:40:50: Writing report file 'fn_jira.components_py.3.html'
2024-02-14T20:40:50.2929510Z 2024-02-14T20:40:50: Creating report 5/15 (Assembly: fn_jira.poller, Class: __init__.py)
2024-02-14T20:40:50.2934476Z 2024-02-14T20:40:50: Writing report file 'fn_jira.poller_py.html'
2024-02-14T20:40:50.2970557Z 2024-02-14T20:40:50: Creating report 6/15 (Assembly: fn_jira.poller, Class: configure_tab.py)
2024-02-14T20:40:50.2975663Z 2024-02-14T20:40:50: Writing report file 'fn_jira.poller_py.2.html'
2024-02-14T20:40:50.2992717Z 2024-02-14T20:40:50: Creating report 7/15 (Assembly: fn_jira.poller, Class: poller.py)
2024-02-14T20:40:50.3002979Z 2024-02-14T20:40:50: Writing report file 'fn_jira.poller_py.3.html'
2024-02-14T20:40:50.3060534Z 2024-02-14T20:40:50: Creating report 8/15 (Assembly: fn_jira.util, Class: __init__.py)
2024-02-14T20:40:50.3065586Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.html'
2024-02-14T20:40:50.3087733Z 2024-02-14T20:40:50: Creating report 9/15 (Assembly: fn_jira.util, Class: config.py)
2024-02-14T20:40:50.3093636Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.2.html'
2024-02-14T20:40:50.3123177Z 2024-02-14T20:40:50: Creating report 10/15 (Assembly: fn_jira.util, Class: customize.py)
2024-02-14T20:40:50.3127679Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.3.html'
2024-02-14T20:40:50.3147459Z 2024-02-14T20:40:50: Creating report 11/15 (Assembly: fn_jira.util, Class: helper.py)
2024-02-14T20:40:50.3152425Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.4.html'
2024-02-14T20:40:50.3172982Z 2024-02-14T20:40:50: Creating report 12/15 (Assembly: fn_jira.util, Class: jira.py)
2024-02-14T20:40:50.3178737Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.5.html'
2024-02-14T20:40:50.3199591Z 2024-02-14T20:40:50: Creating report 13/15 (Assembly: fn_jira.util, Class: regex.py)
2024-02-14T20:40:50.3204813Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.6.html'
2024-02-14T20:40:50.3221544Z 2024-02-14T20:40:50: Creating report 14/15 (Assembly: fn_jira.util, Class: selftest.py)
2024-02-14T20:40:50.3236868Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.7.html'
2024-02-14T20:40:50.3254581Z 2024-02-14T20:40:50: Creating report 15/15 (Assembly: fn_jira.util, Class: typings.py)
2024-02-14T20:40:50.3259054Z 2024-02-14T20:40:50: Writing report file 'fn_jira.util_py.8.html'
2024-02-14T20:40:50.3277826Z 2024-02-14T20:40:50: Creating summary
2024-02-14T20:40:50.3282870Z 2024-02-14T20:40:50: Writing report file '/home/vsts/work/_temp/3d03ace6-7b7c-4733-a6c5-33934a319ab6/index.html'
2024-02-14T20:40:50.3346066Z ##[debug]Parser.GenerateHTMLReport: Creating summary file directory: /home/vsts/work/_temp/3d03ace6-7b7c-4733-a6c5-33934a319ab6/Summary_9db7163d
2024-02-14T20:40:50.3346949Z ##[debug]Parser.GenerateHTMLReport: Copying summary file /home/vsts/work/1/s/coverage.xml
2024-02-14T20:40:50.3350515Z ##[debug]PERF : Parser.ReportGeneration : took 55.0911 ms.
2024-02-14T20:40:50.3353997Z ##[debug]ReportGeneratorTool.GetCoverageSummary: Generating coverage summary for the coverage files.
2024-02-14T20:40:50.3362155Z ##[debug]Publishing code coverage summary supported
2024-02-14T20:40:50.3385822Z Publishing coverage summary data.
2024-02-14T20:40:50.3386564Z line- 664 of 852 covered.
2024-02-14T20:40:50.4238873Z ##[debug]PERF : AzurePipelinesPublisher.UploadSummary : took 74.0404 ms.
2024-02-14T20:40:50.4241473Z ##[debug]PERF : CoverageProcesser.PublishCoverageSummary : took 87.6767 ms.
2024-02-14T20:40:50.4256267Z Publishing file coverage data.
2024-02-14T20:40:50.4389184Z ##[debug]PublisherTraceSource Verbose: 0 : 
2024-02-14T20:40:50.4390124Z ##[debug]Entering UploadTestBuildLogAsync
2024-02-14T20:40:50.5615975Z ##[debug]PublisherTraceSource Verbose: 0 : 
2024-02-14T20:40:50.5619191Z ##[debug]Blob name: Intermediate/639e5b07-a329-4137-99af-fa6a13325f712167500eafdf50a-394f-4fef-a4af-bcbd2ba25490.cjson .
2024-02-14T20:40:51.0102897Z ##[debug]PublisherTraceSource Verbose: 0 : 
2024-02-14T20:40:51.0103784Z ##[debug]File uploaded successsfully on LogStore 639e5b07-a329-4137-99af-fa6a13325f712167500.cjson
2024-02-14T20:40:51.0136361Z ##[debug]PERF : AzurePipelinesPublisher.UploadFileCoverage : took 576.4945 ms.
2024-02-14T20:40:51.0137168Z ##[debug]PERF : CoverageProcesser.PublishFileCoverage : took 589.5573 ms.
2024-02-14T20:40:51.0257981Z 
2024-02-14T20:40:51.0272634Z ##[debug]Exit code 0 received from tool '/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console'
2024-02-14T20:40:51.0274798Z ##[debug]STDIO streams have closed for tool '/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.235.1/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console'
2024-02-14T20:40:51.0457938Z ##[section]Finishing: PublishCodeCoverageResults </pre>
</details>

<details>
  <summary>SUCCESSFUL RUN (Don't have debug annotations: can't have them appear in the past)</summary>
  <pre>2024-01-31T17:46:29.8400986Z ##[section]Starting: PublishCodeCoverageResults
2024-01-31T17:46:29.8405629Z ==============================================================================
2024-01-31T17:46:29.8405768Z Task         : Publish code coverage results v2
2024-01-31T17:46:29.8405836Z Description  : Publish any of the code coverage results from a build
2024-01-31T17:46:29.8405922Z Version      : 2.234.0
2024-01-31T17:46:29.8405976Z Author       : Microsoft Corporation
2024-01-31T17:46:29.8406052Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-code-coverage-results
2024-01-31T17:46:29.8406160Z ==============================================================================
2024-01-31T17:46:30.0106314Z Found 1 result(s) matching pattern: coverage.xml
2024-01-31T17:46:30.0109357Z /home/vsts/work/1/s/coverage.xml
2024-01-31T17:46:30.0202644Z [command]/home/vsts/work/_tasks/PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97/2.234.0/node_modules/coveragepublisher/CoveragePublisher/linux-x64/CoveragePublisher.Console /home/vsts/work/1/s/coverage.xml --reportDirectory /home/vsts/work/_temp/c6a7762c-ff32-4bab-abeb-aad0bea9fe57
2024-01-31T17:46:31.4242349Z Publishing file coverage data.
2024-01-31T17:46:31.4438167Z 2024-01-31T17:46:31: Loading report '/home/vsts/work/1/s/coverage.xml' 1/1 in memory
2024-01-31T17:46:31.4660319Z 2024-01-31T17:46:31: Preprocessing report
2024-01-31T17:46:31.4684568Z 2024-01-31T17:46:31: Initiating parser for Cobertura
2024-01-31T17:46:31.4778700Z 2024-01-31T17:46:31: Current Assembly: fn_jira
2024-01-31T17:46:31.4914311Z 2024-01-31T17:46:31: Current Assembly: fn_jira.components
2024-01-31T17:46:31.4955362Z 2024-01-31T17:46:31: Current Assembly: fn_jira.poller
2024-01-31T17:46:31.4962804Z 2024-01-31T17:46:31: Current Assembly: fn_jira.util
2024-01-31T17:46:31.4988694Z 2024-01-31T17:46:31: Finished parsing '/home/vsts/work/1/s/coverage.xml' 1/1
2024-01-31T17:46:31.4989175Z 2024-01-31T17:46:31: Parsing of 1 files completed
2024-01-31T17:46:31.4989531Z 2024-01-31T17:46:31: Starting merging result 1
2024-01-31T17:46:31.4999808Z 2024-01-31T17:46:31: Finished merging result 1
2024-01-31T17:46:31.5254905Z 2024-01-31T17:46:31: Initializing report builders for report types: HtmlInline_AzurePipelines
2024-01-31T17:46:31.5319186Z 2024-01-31T17:46:31: Analyzing 15 classes
2024-01-31T17:46:31.5321994Z 2024-01-31T17:46:31: Creating report 1/15 (Assembly: fn_jira, Class: __init__.py)
2024-01-31T17:46:31.5468403Z 2024-01-31T17:46:31: Writing report file 'fn_jira_py.html'
2024-01-31T17:46:31.5651673Z 2024-01-31T17:46:31: Creating report 2/15 (Assembly: fn_jira.components, Class: __init__.py)
2024-01-31T17:46:31.5660798Z 2024-01-31T17:46:31: Writing report file 'fn_jira.components_py.html'
2024-01-31T17:46:31.5678791Z 2024-01-31T17:46:31: Creating report 3/15 (Assembly: fn_jira.components, Class: jira_create_comment.py)
2024-01-31T17:46:31.5687538Z 2024-01-31T17:46:31: Writing report file 'fn_jira.components_py.2.html'
2024-01-31T17:46:31.5769257Z 2024-01-31T17:46:31: Creating report 4/15 (Assembly: fn_jira.components, Class: jira_transition_issue.py)
2024-01-31T17:46:31.5776866Z 2024-01-31T17:46:31: Writing report file 'fn_jira.components_py.3.html'
2024-01-31T17:46:31.5787188Z 2024-01-31T17:46:31: Creating report 5/15 (Assembly: fn_jira.poller, Class: __init__.py)
2024-01-31T17:46:31.5793849Z 2024-01-31T17:46:31: Writing report file 'fn_jira.poller_py.html'
2024-01-31T17:46:31.5801005Z 2024-01-31T17:46:31: Creating report 6/15 (Assembly: fn_jira.poller, Class: configure_tab.py)
2024-01-31T17:46:31.5808099Z 2024-01-31T17:46:31: Writing report file 'fn_jira.poller_py.2.html'
2024-01-31T17:46:31.5816063Z 2024-01-31T17:46:31: Creating report 7/15 (Assembly: fn_jira.poller, Class: poller.py)
2024-01-31T17:46:31.5825068Z 2024-01-31T17:46:31: Writing report file 'fn_jira.poller_py.3.html'
2024-01-31T17:46:31.5870045Z 2024-01-31T17:46:31: Creating report 8/15 (Assembly: fn_jira.util, Class: __init__.py)
2024-01-31T17:46:31.5878372Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.html'
2024-01-31T17:46:31.5886317Z 2024-01-31T17:46:31: Creating report 9/15 (Assembly: fn_jira.util, Class: config.py)
2024-01-31T17:46:31.5893343Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.2.html'
2024-01-31T17:46:31.5936128Z 2024-01-31T17:46:31: Creating report 10/15 (Assembly: fn_jira.util, Class: customize.py)
2024-01-31T17:46:31.5936466Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.3.html'
2024-01-31T17:46:31.5941705Z 2024-01-31T17:46:31: Creating report 11/15 (Assembly: fn_jira.util, Class: helper.py)
2024-01-31T17:46:31.5947132Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.4.html'
2024-01-31T17:46:31.5960015Z 2024-01-31T17:46:31: Creating report 12/15 (Assembly: fn_jira.util, Class: jira.py)
2024-01-31T17:46:31.5966492Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.5.html'
2024-01-31T17:46:31.5979197Z 2024-01-31T17:46:31: Creating report 13/15 (Assembly: fn_jira.util, Class: regex.py)
2024-01-31T17:46:31.5985390Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.6.html'
2024-01-31T17:46:31.5992575Z 2024-01-31T17:46:31: Creating report 14/15 (Assembly: fn_jira.util, Class: selftest.py)
2024-01-31T17:46:31.5998657Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.7.html'
2024-01-31T17:46:31.6007524Z 2024-01-31T17:46:31: Creating report 15/15 (Assembly: fn_jira.util, Class: typings.py)
2024-01-31T17:46:31.6013667Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.8.html'
2024-01-31T17:46:31.6026256Z 2024-01-31T17:46:31: Creating summary
2024-01-31T17:46:31.6066831Z 2024-01-31T17:46:31: Writing report file '/home/vsts/work/_temp/c6a7762c-ff32-4bab-abeb-aad0bea9fe57/index.html'
2024-01-31T17:46:31.6421999Z 2024-01-31T17:46:31: Initializing report builders for report types: HtmlInline_AzurePipelines
2024-01-31T17:46:31.6427139Z 2024-01-31T17:46:31: Analyzing 15 classes
2024-01-31T17:46:31.6470332Z 2024-01-31T17:46:31: Creating report 1/15 (Assembly: fn_jira, Class: __init__.py)
2024-01-31T17:46:31.6476272Z 2024-01-31T17:46:31: Writing report file 'fn_jira_py.html'
2024-01-31T17:46:31.6482102Z 2024-01-31T17:46:31: Creating report 2/15 (Assembly: fn_jira.components, Class: __init__.py)
2024-01-31T17:46:31.6484854Z 2024-01-31T17:46:31: Writing report file 'fn_jira.components_py.html'
2024-01-31T17:46:31.6491485Z 2024-01-31T17:46:31: Creating report 3/15 (Assembly: fn_jira.components, Class: jira_create_comment.py)
2024-01-31T17:46:31.6496160Z 2024-01-31T17:46:31: Writing report file 'fn_jira.components_py.2.html'
2024-01-31T17:46:31.6513313Z 2024-01-31T17:46:31: Creating report 4/15 (Assembly: fn_jira.components, Class: jira_transition_issue.py)
2024-01-31T17:46:31.6513679Z 2024-01-31T17:46:31: Writing report file 'fn_jira.components_py.3.html'
2024-01-31T17:46:31.6532182Z 2024-01-31T17:46:31: Creating report 5/15 (Assembly: fn_jira.poller, Class: __init__.py)
2024-01-31T17:46:31.6532516Z 2024-01-31T17:46:31: Writing report file 'fn_jira.poller_py.html'
2024-01-31T17:46:31.6546444Z 2024-01-31T17:46:31: Creating report 6/15 (Assembly: fn_jira.poller, Class: configure_tab.py)
2024-01-31T17:46:31.6546805Z 2024-01-31T17:46:31: Writing report file 'fn_jira.poller_py.2.html'
2024-01-31T17:46:31.6571998Z 2024-01-31T17:46:31: Creating report 7/15 (Assembly: fn_jira.poller, Class: poller.py)
2024-01-31T17:46:31.6572358Z 2024-01-31T17:46:31: Writing report file 'fn_jira.poller_py.3.html'
2024-01-31T17:46:31.6643053Z 2024-01-31T17:46:31: Creating report 8/15 (Assembly: fn_jira.util, Class: __init__.py)
2024-01-31T17:46:31.6646927Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.html'
2024-01-31T17:46:31.6658059Z 2024-01-31T17:46:31: Creating report 9/15 (Assembly: fn_jira.util, Class: config.py)
2024-01-31T17:46:31.6662846Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.2.html'
2024-01-31T17:46:31.6675415Z 2024-01-31T17:46:31: Creating report 10/15 (Assembly: fn_jira.util, Class: customize.py)
2024-01-31T17:46:31.6680225Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.3.html'
2024-01-31T17:46:31.6693009Z 2024-01-31T17:46:31: Creating report 11/15 (Assembly: fn_jira.util, Class: helper.py)
2024-01-31T17:46:31.6697959Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.4.html'
2024-01-31T17:46:31.6711900Z 2024-01-31T17:46:31: Creating report 12/15 (Assembly: fn_jira.util, Class: jira.py)
2024-01-31T17:46:31.6717070Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.5.html'
2024-01-31T17:46:31.6735987Z 2024-01-31T17:46:31: Creating report 13/15 (Assembly: fn_jira.util, Class: regex.py)
2024-01-31T17:46:31.6738684Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.6.html'
2024-01-31T17:46:31.6758293Z 2024-01-31T17:46:31: Creating report 14/15 (Assembly: fn_jira.util, Class: selftest.py)
2024-01-31T17:46:31.6761031Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.7.html'
2024-01-31T17:46:31.6783065Z 2024-01-31T17:46:31: Creating report 15/15 (Assembly: fn_jira.util, Class: typings.py)
2024-01-31T17:46:31.6787939Z 2024-01-31T17:46:31: Writing report file 'fn_jira.util_py.8.html'
2024-01-31T17:46:31.6802437Z 2024-01-31T17:46:31: Creating summary
2024-01-31T17:46:31.6805150Z 2024-01-31T17:46:31: Writing report file '/home/vsts/work/_temp/c6a7762c-ff32-4bab-abeb-aad0bea9fe57/index.html'
2024-01-31T17:46:31.6893360Z Publishing coverage summary data.
2024-01-31T17:46:31.6896791Z line- 671 of 860 covered.
2024-01-31T17:46:31.8234842Z Publishing file coverage data.
2024-01-31T17:46:32.6035872Z Publishing code coverage HTML report.
2024-01-31T17:46:32.6098987Z Uploading 19 files.
2024-01-31T17:46:35.9098353Z File upload succeed.
2024-01-31T17:46:35.9769386Z Published '/home/vsts/work/_temp/c6a7762c-ff32-4bab-abeb-aad0bea9fe57' as artifact 'Code Coverage Report_2134692'.
2024-01-31T17:46:35.9995577Z ##[section]Finishing: PublishCodeCoverageResults</pre>
</details>

### Repro steps

```yml
trigger:
  - main
pool:
  vmImage: 'ubuntu-22.04'
steps:
  - script: pytest --cov=. --cov-report=xml
    displayName: 'Pytest'
  - task: PublishCodeCoverageResults@2
    inputs:
      summaryFileLocation: coverage.xml
  # Download coverage report to trigger an error
  - task: DownloadPipelineArtifact@2
    inputs:
      artifact: Code Coverage Report_$(Build.BuildId)
      patterns: '**/coverage.xml'
      path: .
```

**Comment by GavinToddlds** [comment]
I have a run from yesterday using 2.235.0 that published the code coverage report and then all runs today using the same version 2.235.0 are failing

**Comment by GavinToddlds** [comment]
The 2.236.0 release didn't fix the issue. Is this being looked at?

**Comment by SemaphoreSlim1** [comment]
For anybody that's paying attention, can confirm that this is affecting me and my organization as well

**Comment by bunnu** [comment]
Same here too.

**Comment by GavinToddlds** [comment]
@rasunkar Is this something we could get looked at?

**Comment by TGrochowski87** [comment]
Hello. Are there any updates on this? My organization is also affected.

**Comment by carboneater** [comment]
I also filed a ticket on their devops feedback form due to the obvious lack of M$ answers here...
They said they triaged it... crickets ever since...

You still may want to vote for it though...
https://developercommunity.visualstudio.com/t/REGRESSION:-PublishCodeCoverageResults/10620444

**Comment by GavinToddlds** [comment]
Is this not a regression and something they intended to change at this point?

**Comment by JtMotoX** [comment]
The major issues this is causing us is that we have approval stages within our pipelines that allows application leads and operations to approve/decline the build. Without the results being published as an artifact, the application leads can no longer make a determination if they should approve or decline.

One more thing to point out. The v1 "Code Coverages" tab used to show the results immediately, allowing the application leads and operations to determine if they should Approve or Decline the continuation of the build. The new v2 does not show the "Code Coverage" results until the build is complete and shows "Build pipeline is queued. Results will be generated once build is complete.".  How are the application leads and operations supposed to make a determination if they should approve the build if they can't even view the results?

**Comment by carboneater** [comment]
Plot twist, version 2.240.1 seems to have fixed the issue on my side.
Do you guys have the same resolution?

**Comment by bunnu** [comment]
Same here. Since ~12th of June the pipeline artifacts include the coverage report again.

**Comment by GavinToddlds** [comment]
I'm seeing it again as well. Thank goodness!

**Comment by cacti77** [comment]
Yup, with 2.240.1 the warning's gone, but the Code Coverage tab now displays the results and the coverage report is uploaded to the pipeline artifacts, as expected. So I've finally upgraded from v1 to v2 today.

**Comment by carboneater** [comment]
Closing as someone seems to have fixed it silently in v2.240.1

Here's hoping it stays fixed...
